### ISSUE TITLE
Incorrect display of variable products on FB-Shop

### ISSUE THREAD

**Issue by AndreasMe** [facebookincubator/facebook-for-woocommerce#114]
Hello,
after I've synced my products to FB-shop, variable products showing one of the variations with it’s price instead of ‘choose an variation’. How could I fix that?

And if s.o. selects on of the variations than in product details a  detail of another variation is showing up.

regards, Andreas

**Comment by mengyingdu** [comment]
Hi,

If a product is variable product, we will default show the one you set in Woo product data 'default variation value'. If you didn't set it, we will default show the last one added in this group. 

We will consider to add this feature based on feature request and priority. 

Thanks.

**Comment by mengyingdu** [comment]
As for the variation description, did you set different description in each variation of parent product data 'variations' panel? I am trying to repro now. Thanks.

**Comment by AndreasMe** [comment]
Thank you for answering,
Yes, I have different descriptions for each variation.

**Comment by AndreasMe** [comment]
And there is no default variation

**Comment by AndreasMe** [comment]
Here are sreenshots:
![fb02](https://user-images.githubusercontent.com/33100951/32026530-17ab808e-b9e5-11e7-8b1b-19dd71e8b8ac.jpeg)
![fb01](https://user-images.githubusercontent.com/33100951/32026531-17c4b6c6-b9e5-11e7-9706-473554fd7d83.jpeg)
![woo-02-](https://user-images.githubusercontent.com/33100951/32026532-17e062cc-b9e5-11e7-901c-b0c4a0bf03c9.jpeg)
![woo-01-](https://user-images.githubusercontent.com/33100951/32026533-17fa4b38-b9e5-11e7-8cd8-0d96f933714f.jpeg)

**Comment by mengyingdu** [comment]
Hi,

Thanks for your screen shot and detailed information. I just added variant description feature today. I checked on different variant product page (redirect from Woo, right side, different product if) and it's works.
However, if we enter from shop->a variable product group, in the popup window->change variant value, variant description can't update now. This is a bug for our shop team and I have already flag this to them.

To sum up:
1. Hope we could get shop team support to get the popup window, variant description feature will be added in new release. (On plugins side, it works now but we need to make the entire process work)
2. As quick solution, I would recommend you to use a general description under 'facebook' panel, description box.

Thanks.

**Comment by AndreasMe** [comment]
Hi,
thanks for your reply. I really don't know where I can find this description box on FB. I'm using fb-businessmanager. Perhaps it is not fine for this?

regards

**Comment by mengyingdu** [comment]
Hi,

This description FB box is in WooCommerce product data page like your last screen shot. I also attached another screen shot to help you find this:
![facebook description](https://user-images.githubusercontent.com/24535640/32067159-0e582c06-ba37-11e7-9add-5dee522a4d7c.jpeg)

**Comment by AndreasMe** [comment]
Hi,
ah ok, Idid not recognize that.

Thank you!!!

**Comment by mengyingdu** [comment]
Duplicate of #111

**Comment by mengyingdu** [comment]
Hi @AndreasMe ,

Some good news to update: our shop team just fixed the shop bug!
To be clear,
if we enter from shop->a variable product group, in the popup window->change variant value, variant description WILL update now.

**Comment by PeterHygge** [comment]
Hi

In Woocommerce my default value for variables is blank, so the products will display the price range.
On Facebook Shop this seems to default to the highest value, which may put people off buying if they think it is too expensive.

**Comment by mengyingdu** [comment]
Hi @PeterHygge ,

Do you mean the price in a new window when you click on a product in your shop? 
If you don't set default, currently, it will link to the last product of this variable product group. If you set default, it will link to default.

Thanks.

**Comment by mengyingdu** [comment]
Hi @PeterHygge ,

Close this issue for now since there is no update info from your side.
Feel free to open a new one if you still have this problem and need more help.
Thanks!
